python/pyspark/sql/connect/dataframe.py

```diff
@@ -137,6 +137,14 @@ def __init__(
         # by __repr__ and _repr_html_ while eager evaluation opens.
         self._support_repr_html = False
         self._cached_schema: Optional[StructType] = None
+        self._cached_remote_relation_ids: List[str] = []
```

Comment:
nit: why list not optional?

Reply:
We can make it optional, and assign `None` for its initial value too